address/src/lib.rs

```diff
@@ -72,6 +72,14 @@ impl<const PREFIX: u8> BaseAddress<PREFIX> {
             .expect("MAX_ADDR_LEN is a u8 so this can't fail")
     }
 
+    /// Returns the address bytes prefixed with the length of the address.
+    pub fn prefix_len_bytes(&self) -> Vec<u8> {
+        let len = self.len();
+        let mut prefixed = vec![len];
+        prefixed.extend_from_slice(&self.0);
```

Comment:
What about `[&[prefixed], &self.0].concat()`?

Reply:
To make it work I have to do `[prefix, self.0.clone()].concat()`. This could require two allocations of `self.0`. How about:
```
       let len = self.len();
        let mut prefixed = Vec::with_capacity(len as usize + 1);
        prefixed.push(len);
        prefixed.extend_from_slice(&self.0);
        prefixed
        ```

Reply:
Check [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=aa7d5b668d1dd009048c53022a7fd036) playground.

Reply:
Went with:
```
let len = self.len();
[&[len], self.0.as_slice()].concat()
```
Suggested by @NYBACHOK